In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# 1. **DATA INSPECTION**
## a. Load the dataset into a pandas DataFrame and display the first 10 rows.

In [3]:
df = pd.read_csv(r"C:\Users\jinay\Downloads\IMdB_India_Top250.csv")
df.head(10)

FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\jinay\\Downloads\\IMdB_India_Top250.csv'

## b. Use the info() function to get an overview of the dataset, including data types and non-null counts.

In [27]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 250 entries, 0 to 249
Data columns (total 13 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Movie name             250 non-null    object 
 1   Year of release        250 non-null    int64  
 2   Watch  hour            250 non-null    object 
 3   Rating                 250 non-null    float64
 4   Ratedby                250 non-null    object 
 5   Film Industry          250 non-null    object 
 6   Genre                  250 non-null    object 
 7   Director               250 non-null    object 
 8   Box office collection  250 non-null    object 
 9   User reviews           250 non-null    int64  
 10  Awards                 250 non-null    object 
 11  Description            250 non-null    object 
 12  Streaming platform     250 non-null    object 
dtypes: float64(1), int64(2), object(10)
memory usage: 25.5+ KB


## c. Display the summary statistics for numerical columns using describe().

In [28]:
df.describe()

Year of release      Rating  User reviews
count       250.000000  250.000000     250.00000
mean       2008.388000    8.190400     222.18400
std          14.708597    0.252513     388.15949
min        1955.000000    7.700000      10.00000
25%        2003.000000    8.000000      39.00000
50%        2013.000000    8.150000      94.50000
75%        2018.000000    8.300000     231.50000
max        2024.000000    9.100000    3300.00000

# 2. **DATA CLEANING**
## a. Identify and handle any missing values in the dataset. Drop rows or fill missing values as appropriate.

In [1]:
df.isnull().sum()
#As there are no null values, there is no need to handle missing values.

NameError: name 'df' is not defined

## b. Remove any duplicate rows from the dataset. Ensure that the columns related to rankings, years, or ratings are in the correct numerical format.

In [30]:
df.duplicated().count()
#There are also no duplicate values. So no need to drop.

np.int64(250)

In [31]:
print(df.dtypes)
# Rankings, Years and Ratings are in the correct numerical format.

Movie name                object
Year of release            int64
Watch  hour               object
Rating                   float64
Ratedby                   object
Film Industry             object
Genre                     object
Director                  object
Box office collection     object
User reviews               int64
Awards                    object
Description               object
Streaming platform        object
dtype: object


# 3. **FILTERING AND SORTING**
## a. Filter the dataset to display only movies released after the year 2000.

In [32]:
df[df['Year of release']>2000]

Movie name  Year of release        Watch  hour   Rating  \
0                   12th Fail             2023  2 hours 27 minutes     8.9   
2                    Maharaja             2024  2 hours 30 minutes     8.6   
5                  Anbe Sivam             2003  2 hours 40 minutes     8.6   
6           Pariyerum Perumal             2018  2 hours 34 minutes     8.7   
7                    3 Idiots             2009  2 hours 50 minutes     8.4   
..                        ...              ...                 ...     ...   
244          Angamaly Diaries             2017  2 hours 12 minutes     7.9   
246  Hazaaron Khwaishein Aisi             2003            2 hours      7.9   
247                    Ghilli             2004  2 hours 40 minutes     8.1   
248  Mukundan Unni Associates             2022   2 hours 8 minutes     7.9   
249                Happy Days             2007  2 hours 33 minutes     7.9   

    Ratedby          Film Industry                     Genre  \
0      126K      Bollywood (Hindi)          Drama, Biography   
2       37K      Kollywood (Tamil)              Crime, Drama   
5       26K      Kollywood (Tamil)             Drama, Comedy   
6       19K      Kollywood (Tamil)             Drama, Social   
7      441K      Bollywood (Hindi)             Comedy, Drama   
..      ...                    ...                       ...   
244    6.8K  Mollywood (Malayalam)              Crime, Drama   
246    5.3K      Bollywood (Hindi)  Political Drama, Romance   
247     17K      Kollywood (Tamil)             Action, Drama   
248    6.2K  Mollywood (Malayalam)             Satire, Drama   
249    6.3K     Tollywood (Telugu)    Drama, Romance, Comedy   

                 Director Box office collection  User reviews  \
0      Vidhu Vinod Chopra           $138,288.00           945   
2     Nithilan Saminathan           $975,543.00           370   
5               Sundar C.                   NIL           115   
6           Mari Selvaraj                   NIL           169   
7         Rajkumar Hirani        $60,262,836.00          1000   
..                    ...                   ...           ...   
244  Lijo Jose Pellissery           $661,915.00            39   
246         Sudhir Mishra                   NIL            44   
247               Dharani           $163,961.00            47   
248  Abhinav Sunder Nayak            $44,990.00            62   
249        Sekhar Kammula           $361,445.78            16   

                       Awards  \
0    23 wins & 32 nominations   
2               2 nominations   
5      2 wins & 3 nominations   
6     11 wins & 5 nominations   
7    64 wins & 30 nominations   
..                        ...   
244    2 wins & 6 nominations   
246    8 wins & 9 nominations   
247    2 wins & 2 nominations   
248                     1 win   
249                    7 wins   

                                           Description  \
0    The real-life story of IPS Officer Manoj Kumar...   
2    A barber seeks vengeance after his home is bur...   
5    Two men, one young and arrogant, the other dam...   
6    A law student from a lower caste begins a frie...   
7    Two friends are searching for their long lost ...   
..                                                 ...   
244  Vincent Pepe who wanted to be a powerful leade...   
246  Set in the backdrop of Indian Emergency 1975, ...   
247  Velu, an aspiring kabaddi player, goes to Madu...   
248  Advocate Mukundan Unni, played by Vineeth Sree...   
249  A group of eight youngsters from an engineerin...   

              Streaming platform  
0                        SonyLIV  
2                        Netflix  
5                Disney+ Hotstar  
6             Amazon Prime Video  
7    Netflix, Amazon Prime Video  
..                           ...  
244                      Netflix  
246           Amazon Prime Video  
247              Disney+ Hotstar  
248              Disney+ Hotstar  
249           Amazon Prime Video  

[202 rows x 13 co

## b. Sort the DataFrame by IMDB Rating in descending order.

In [33]:
df.sort_values(by='Rating',ascending=False)

Movie name  Year of release        Watch  hour   Rating  \
27              Mayabazar             1957  3 hours 12 minutes     9.1   
17              Sandesham             1991  2 hours 18 minutes     9.0   
14               Kireedam             1989   2 hours 4 minutes     8.9   
0               12th Fail             2023  2 hours 27 minutes     8.9   
23              Natsamrat             2016  2 hours 46 minutes     8.8   
..                    ...              ...                 ...     ...   
236      Stanley Ka Dabba             2011   1 hour 36 minutes     7.8   
239            Veer-Zaara             2004  3 hours 12 minutes     7.8   
237  Nayak: The Real Hero             2001   3 hours 7 minutes     7.8   
245             Mr. India             1987  2 hours 59 minutes     7.7   
243           Vicky Donor             2012   2 hours 6 minutes     7.7   

    Ratedby          Film Industry                     Genre  \
27     5.9K     Tollywood (Telugu)     Fantasy, Mythological   
17     5.4K  Mollywood (Malayalam)  Political Satire, Comedy   
14     8.8K  Mollywood (Malayalam)             Drama, Family   
0      126K      Bollywood (Hindi)          Drama, Biography   
23     7.3K         Marathi Cinema            Drama, Tragedy   
..      ...                    ...                       ...   
236    7.7K      Bollywood (Hindi)             Drama, Family   
239     58K      Bollywood (Hindi)            Romance, Drama   
237     18K      Bollywood (Hindi)   Action, Political Drama   
245     17K      Bollywood (Hindi)    Action, Sci-Fi, Family   
243     46K      Bollywood (Hindi)             Comedy, Drama   

                 Director Box office collection  User reviews  \
27   Kadiri Venkata Reddy                   NIL            39   
17       Sathyan Anthikad                   NIL            17   
14           Sibi Malayil           $181,001.00            33   
0      Vidhu Vinod Chopra           $138,288.00           945   
23       Mahesh Manjrekar               $633.00            72   
..                    ...                   ...           ...   
236           Amole Gupte         $1,241,102.00            44   
239           Yash Chopra         $6,887,221.00           197   
237            S. Shankar                   NIL            20   
245         Shekhar Kapur                   NIL            25   
243        Shoojit Sircar         $6,456,358.00            82   

                       Awards  \
27                        NIL   
17                     2 wins   
14                      1 win   
0    23 wins & 32 nominations   
23     2 wins & 5 nominations   
..                        ...   
236    4 wins & 6 nominations   
239  30 wins & 42 nominations   
237             4 nominations   
245                     1 win   
243  43 wins & 37 nominations   

                                           Description  \
27   Balarama promises Subhadra to get his daughter...   
17   A satire on contemporary Kerala politics where...   
14   The life of a young man turns upside down when...   
0    The real-life story of IPS Officer Manoj Kumar...   
23   After he retires, an aging theatre actor and h...   
..                                                 ...   
236  A school-teacher, who forces children to share...   
239  "Veer-Zaara" is a saga of love, separation, co...   
237  A man accepts a challenge by the chief ministe...   
245  A poor but big-hearted man takes orphans into ...   
243  A man is brought in by an infertility doctor t...   

              Streaming platform  
27   Amazon Prime Video, YouTube  
17            Amazon Prime Video  
14            Amazon Prime Video  
0                        SonyLIV  
23                          Zee5  
..                           ...  
236              Disney+ Hotstar  
239           Amazon Prime Video  
237           Amazon Prime Video  
245     Zee5, Amazon Prime Video  
243  Amazon Prime Video, Netflix  

[250 rows x 13 columns]

## c. Create a new DataFrame that contains only movies with an IMDB rating greater than 8.5.

In [34]:
df2 = df[df['Rating']>8.5]
df2.head()

Movie name  Year of release        Watch  hour   Rating Ratedby  \
0          12th Fail             2023  2 hours 27 minutes     8.9    126K   
2           Maharaja             2024  2 hours 30 minutes     8.6     37K   
3            Nayakan             1987  2 hours 25 minutes     8.7     25K   
5         Anbe Sivam             2003  2 hours 40 minutes     8.6     26K   
6  Pariyerum Perumal             2018  2 hours 34 minutes     8.7     19K   

       Film Industry             Genre             Director  \
0  Bollywood (Hindi)  Drama, Biography   Vidhu Vinod Chopra   
2  Kollywood (Tamil)      Crime, Drama  Nithilan Saminathan   
3  Kollywood (Tamil)      Crime, Drama          Mani Ratnam   
5  Kollywood (Tamil)     Drama, Comedy            Sundar C.   
6  Kollywood (Tamil)     Drama, Social        Mari Selvaraj   

  Box office collection  User reviews                    Awards  \
0           $138,288.00           945  23 wins & 32 nominations   
2           $975,543.00           370             2 nominations   
3           $120,481.93           237     7 wins & 1 nomination   
5                   NIL           115    2 wins & 3 nominations   
6                   NIL           169   11 wins & 5 nominations   

                                         Description  \
0  The real-life story of IPS Officer Manoj Kumar...   
2  A barber seeks vengeance after his home is bur...   
3  A common man's struggles against a corrupt pol...   
5  Two men, one young and arrogant, the other dam...   
6  A law student from a lower caste begins a frie...   

            Streaming platform  
0                      SonyLIV  
2                      Netflix  
3  Amazon Prime Video, YouTube  
5              Disney+ Hotstar  
6           Amazon Prime Video

# 4. **GROUP BY OPERATIONS**
## a. Group the movies by the Year column and calculate the average IMDB rating for each year.

In [35]:
df3 = df.groupby('Year of release')
df3.first().head()

Movie name        Watch  hour   Rating Ratedby  \
Year of release                                                         
1955              Pather Panchali   2 hours 5 minutes     8.2     39K   
1956                    Aparajito   1 hour 50 minutes     8.2     16K   
1957                    Mayabazar  3 hours 12 minutes     9.1    5.9K   
1958               The Music Room   1 hour 35 minutes     7.9    6.8K   
1959             The World of Apu   1 hour 45 minutes     8.4     17K   

                      Film Industry                  Genre  \
Year of release                                              
1955                 Bengali Cinema                  Drama   
1956                 Bengali Cinema                  Drama   
1957             Tollywood (Telugu)  Fantasy, Mythological   
1958                 Bengali Cinema                  Drama   
1959                 Bengali Cinema                  Drama   

                             Director Box office collection  User reviews  \
Year of release                                                             
1955                     Satyajit Ray           $135,342.00           204   
1956                     Satyajit Ray           $134,241.00            59   
1957             Kadiri Venkata Reddy                   NIL            39   
1958                     Satyajit Ray             $3,247.00            42   
1959                     Satyajit Ray           $134,241.00            62   

                                        Awards  \
Year of release                                  
1955             11 wins & 2 nominations total   
1956              6 wins & 2 nominations total   
1957                                       NIL   
1958                    2 wins & 2 nominations   
1959              4 wins & 2 nominations total   

                                                       Description  \
Year of release                                                      
1955             Impoverished priest Harihar Ray, dreaming of a...   
1956             Following his father's death, a boy leaves hom...   
1957             Balarama promises Subhadra to get his daughter...   
1958             Depicts the end days of a decadent zamindar (l...   
1959             This final installment in Satyajit Ray's Apu T...   

                          Streaming platform  
Year of release                               
1955             Amazon Prime Video, Hoichoi  
1956                      Amazon Prime Video  
1957             Amazon Prime Video, YouTube  
1958                      Amazon Prime Video  
1959             Amazon Prime Video, Hoichoi

## b. Group the data by Director and count the number of movies each director has in the top 250.

In [36]:
df4 = df.groupby('Director')
df4.first().head()

Movie name  Year of release  \
Director                                                          
A.L. Vijay                    Deiva Thirumagal             2011   
A.R. Murugadoss                      Thuppakki             2012   
Aamir Khan                 Like Stars on Earth             2007   
Abhinav Sunder Nayak  Mukundan Unni Associates             2022   
Abhishek Pathak                     Drishyam 2             2022   

                            Watch  hour   Rating Ratedby  \
Director                                                   
A.L. Vijay            2 hours 46 minutes     8.2    6.6K   
A.R. Murugadoss       2 hours 45 minutes     8.1     38K   
Aamir Khan            2 hours 42 minutes     8.3    209K   
Abhinav Sunder Nayak   2 hours 8 minutes     7.9    6.2K   
Abhishek Pathak       2 hours 20 minutes     8.2     46K   

                              Film Industry             Genre  \
Director                                                        
A.L. Vijay                Kollywood (Tamil)     Drama, Family   
A.R. Murugadoss           Kollywood (Tamil)  Action, Thriller   
Aamir Khan                Bollywood (Hindi)     Drama, Family   
Abhinav Sunder Nayak  Mollywood (Malayalam)     Satire, Drama   
Abhishek Pathak       Mollywood (Malayalam)   Thriller, Drama   

                     Box office collection  User reviews  \
Director                                                   
A.L. Vijay                     $846,407.00            49   
A.R. Murugadoss              $2,130,470.00           134   
Aamir Khan                  $21,897,373.00           463   
Abhinav Sunder Nayak            $44,990.00            62   
Abhishek Pathak              $2,973,347.00           179   

                                        Awards  \
Director                                         
A.L. Vijay            13 wins & 13 nominations   
A.R. Murugadoss       14 wins & 17 nominations   
Aamir Khan            28 wins & 18 nominations   
Abhinav Sunder Nayak                     1 win   
Abhishek Pathak        7 wins & 29 nominations   

                                                            Description  \
Director                                                                  
A.L. Vijay            A man with disabilities fights for custody of ...   
A.R. Murugadoss       An army captain is on a mission to track down ...   
Aamir Khan            An eight-year-old boy is thought to be a lazy ...   
Abhinav Sunder Nayak  Advocate Mukundan Unni, played by Vineeth Sree...   
Abhishek Pathak       A gripping tale of an investigation and a fami...   

                            Streaming platform  
Director                                        
A.L. Vijay                  Amazon Prime Video  
A.R. Murugadoss                Disney+ Hotstar  
Aamir Khan            Netflix, Disney+ Hotstar  
Abhinav Sunder Nayak           Disney+ Hotstar  
Abhishek Pathak             Amazon Prime Video

In [37]:
df4.count()

Movie name  Year of release  Watch  hour   Rating  \
Director                                                                  
A.L. Vijay                     1                1             1       1   
A.R. Murugadoss                1                1             1       1   
Aamir Khan                     1                1             1       1   
Abhinav Sunder Nayak           1                1             1       1   
Abhishek Pathak                1                1             1       1   
...                          ...              ...           ...     ...   
Vishal Bhardwaj                3                3             3       3   
Vishnuvardhan                  1                1             1       1   
Vivek Agnihotri                1                1             1       1   
Yash Chopra                    2                2             2       2   
Zoya Akhtar                    2                2             2       2   

                      Ratedby  Film Industry  Genre  Box office collection  \
Director                                                                     
A.L. Vijay                  1              1      1                      1   
A.R. Murugadoss             1              1      1                      1   
Aamir Khan                  1              1      1                      1   
Abhinav Sunder Nayak        1              1      1                      1   
Abhishek Pathak             1              1      1                      1   
...                       ...            ...    ...                    ...   
Vishal Bhardwaj             3              3      3                      3   
Vishnuvardhan               1              1      1                      1   
Vivek Agnihotri             1              1      1                      1   
Yash Chopra                 2              2      2                      2   
Zoya Akhtar                 2              2      2                      2   

                      User reviews  Awards  Description  Streaming platform  
Director                                                                     
A.L. Vijay                       1       1            1                   1  
A.R. Murugadoss                  1       1            1                   1  
Aamir Khan                       1       1            1                   1  
Abhinav Sunder Nayak             1       1            1                   1  
Abhishek Pathak                  1       1            1                   1  
...                            ...     ...          ...                 ...  
Vishal Bhardwaj                  3       3            3                   3  
Vishnuvardhan                    1       1            1                   1  
Vivek Agnihotri                  1       1            1                   1  
Yash Chopra                      2       2            2                   2  
Zoya Akhtar                      2       2            2                   2  

[182 rows x 12 columns]

## c. Find the highest-rated movie for each year by grouping the data by Year and selecting the movie with the highest rating in each group.

In [38]:
df.loc[df.groupby('Year of release')['Rating'].idxmax()]

Movie name  Year of release        Watch  hour   Rating  \
43              Pather Panchali             1955   2 hours 5 minutes     8.2   
38                    Aparajito             1956   1 hour 50 minutes     8.2   
27                    Mayabazar             1957  3 hours 12 minutes     9.1   
196              The Music Room             1958   1 hour 35 minutes     7.9   
4              The World of Apu             1959   1 hour 45 minutes     8.4   
138               Mughal-E-Azam             1960  3 hours 17 minutes     8.1   
120                   Charulata             1964   1 hour 57 minutes     8.1   
54                        Guide             1965   3 hours 3 minutes     8.3   
154                     Padosan             1968  2 hours 37 minutes     8.1   
221             Mera Naam Joker             1970  3 hours 44 minutes     7.9   
128                       Anand             1971   2 hours 2 minutes     8.1   
46                Chupke Chupke             1975  2 hours 26 minutes     8.3   
1                      Gol Maal             1979            2 hours      8.5   
92                       Angoor             1982  2 hours 11 minutes     8.3   
36           Jaane Bhi Do Yaaro             1983  2 hours 12 minutes     8.3   
28                 Nadodikkattu             1987  2 hours 38 minutes     8.8   
58                     Chithram             1988  2 hours 39 minutes     8.6   
14                     Kireedam             1989   2 hours 4 minutes     8.9   
17                    Sandesham             1991  2 hours 18 minutes     9.0   
29                 Thevar Magan             1992  2 hours 25 minutes     8.7   
10            Manichithrathazhu             1993  2 hours 49 minutes     8.7   
151             Andaz Apna Apna             1994  2 hours 40 minutes     8.0   
71                     Spadikam             1995  2 hours 30 minutes     8.6   
134                      Indian             1996   3 hours 5 minutes     8.1   
60                       Iruvar             1997  2 hours 20 minutes     8.4   
44                        Satya             1998  2 hours 50 minutes     8.3   
63                    Mudhalvan             1999  2 hours 49 minutes     8.4   
124               Alai Payuthey             2000  2 hours 36 minutes     8.3   
40         Nuvvu Naaku Nachchav             2001   3 hours 4 minutes     8.8   
65        Kannathil Muthamittal             2002  2 hours 16 minutes     8.3   
5                    Anbe Sivam             2003  2 hours 40 minutes     8.6   
9                  Black Friday             2004  2 hours 23 minutes     8.4   
35                      Anniyan             2005     3 hours 1minute     8.3   
85                 Pudhu Pettai             2006  2 hours 48 minutes     8.4   
16          Like Stars on Earth             2007  2 hours 42 minutes     8.3   
144            Vaaranam Aayiram             2008  2 hours 49 minutes     8.2   
7                      3 Idiots             2009  2 hours 50 minutes     8.4   
83              Aaranya Kaandam             2010   2 hours 6 minutes     8.5   
59     Zindagi Na Milegi Dobara             2011  2 hours 35 minutes     8.2   
51           Gangs of Wasseypur             2012  5 hours 21 minutes     8.2   
34                     Drishyam             2013  2 hours 40 minutes     8.3   
174         Ulidavaru Kandanthe             2014  2 hours 34 minutes     8.4   
24                   Visaaranai             2015   1 hour 57 minutes     8.4   
23                    Natsamrat             2016  2 hours 46 minutes     8.8   
75                 Vikram Vedha             2017  2 hours 27 minutes     8.2   
15            C/o Kancharapalem             2018  2 hours 32 minutes     8.8   
12            Kumbalangi Nights             2019  2 hours 15 minutes     8.5   
21              Soorarai Pottru             2020  2 hours 33 minutes     8.7   
8                         #Home             2021  2 hours 38 minutes     8.8   
11   Rocketry: The Na

# 5. **COLUMN CREATION AND MANIPULATION**
## a. Create a new column named Rating Category that categorizes movies as "Excellent" if the IMDB rating is 9.0 or above, "Good" if between 8.0 and 9.0, and "Average" if below 8.0.

In [39]:
def ratcat(x):
    if x>=9:
        return 'Excellent'
    elif x>=8:
        return 'Good'
    else:
        return 'Average'
    
df['Rating Category'] = df['Rating'].apply(ratcat)
df.head()

Movie name  Year of release        Watch  hour   Rating Ratedby  \
0         12th Fail             2023  2 hours 27 minutes     8.9    126K   
1          Gol Maal             1979            2 hours      8.5     20K   
2          Maharaja             2024  2 hours 30 minutes     8.6     37K   
3           Nayakan             1987  2 hours 25 minutes     8.7     25K   
4  The World of Apu             1959   1 hour 45 minutes     8.4     17K   

       Film Industry             Genre              Director  \
0  Bollywood (Hindi)  Drama, Biography    Vidhu Vinod Chopra   
1  Bollywood (Hindi)            Comedy  Hrishikesh Mukherjee   
2  Kollywood (Tamil)      Crime, Drama   Nithilan Saminathan   
3  Kollywood (Tamil)      Crime, Drama           Mani Ratnam   
4     Bengali Cinema             Drama          Satyajit Ray   

  Box office collection  User reviews                        Awards  \
0           $138,288.00           945      23 wins & 32 nominations   
1                   NIL            48         3 wins & 1 nomination   
2           $975,543.00           370                 2 nominations   
3           $120,481.93           237         7 wins & 1 nomination   
4           $134,241.00            62  4 wins & 2 nominations total   

                                         Description  \
0  The real-life story of IPS Officer Manoj Kumar...   
1  A man's simple lie to secure his job escalates...   
2  A barber seeks vengeance after his home is bur...   
3  A common man's struggles against a corrupt pol...   
4  This final installment in Satyajit Ray's Apu T...   

                  Streaming platform Rating Category  
0                            SonyLIV            Good  
1  Amazon Prime Video, YouTube, Zee5            Good  
2                            Netflix            Good  
3        Amazon Prime Video, YouTube            Good  
4        Amazon Prime Video, Hoichoi            Good

## b. Extract the Year from the movie title (assuming the year is part of the title) and create a new column Extracted Year.

For this question, the movie titles and years are already different columns.

## c. Create a new column that combines the Title and Director into a single string, separated by a hyphen.

In [40]:
df['Combined Column'] = df['Movie name'] + '-' + df['Director']
df

Movie name  Year of release        Watch  hour   Rating  \
0                   12th Fail             2023  2 hours 27 minutes     8.9   
1                    Gol Maal             1979            2 hours      8.5   
2                    Maharaja             2024  2 hours 30 minutes     8.6   
3                     Nayakan             1987  2 hours 25 minutes     8.7   
4            The World of Apu             1959   1 hour 45 minutes     8.4   
..                        ...              ...                 ...     ...   
245                 Mr. India             1987  2 hours 59 minutes     7.7   
246  Hazaaron Khwaishein Aisi             2003            2 hours      7.9   
247                    Ghilli             2004  2 hours 40 minutes     8.1   
248  Mukundan Unni Associates             2022   2 hours 8 minutes     7.9   
249                Happy Days             2007  2 hours 33 minutes     7.9   

    Ratedby          Film Industry                     Genre  \
0      126K      Bollywood (Hindi)          Drama, Biography   
1       20K      Bollywood (Hindi)                    Comedy   
2       37K      Kollywood (Tamil)              Crime, Drama   
3       25K      Kollywood (Tamil)              Crime, Drama   
4       17K         Bengali Cinema                     Drama   
..      ...                    ...                       ...   
245     17K      Bollywood (Hindi)    Action, Sci-Fi, Family   
246    5.3K      Bollywood (Hindi)  Political Drama, Romance   
247     17K      Kollywood (Tamil)             Action, Drama   
248    6.2K  Mollywood (Malayalam)             Satire, Drama   
249    6.3K     Tollywood (Telugu)    Drama, Romance, Comedy   

                 Director Box office collection  User reviews  \
0      Vidhu Vinod Chopra           $138,288.00           945   
1    Hrishikesh Mukherjee                   NIL            48   
2     Nithilan Saminathan           $975,543.00           370   
3             Mani Ratnam           $120,481.93           237   
4            Satyajit Ray           $134,241.00            62   
..                    ...                   ...           ...   
245         Shekhar Kapur                   NIL            25   
246         Sudhir Mishra                   NIL            44   
247               Dharani           $163,961.00            47   
248  Abhinav Sunder Nayak            $44,990.00            62   
249        Sekhar Kammula           $361,445.78            16   

                           Awards  \
0        23 wins & 32 nominations   
1           3 wins & 1 nomination   
2                   2 nominations   
3           7 wins & 1 nomination   
4    4 wins & 2 nominations total   
..                            ...   
245                         1 win   
246        8 wins & 9 nominations   
247        2 wins & 2 nominations   
248                         1 win   
249                        7 wins   

                                           Description  \
0    The real-life story of IPS Officer Manoj Kumar...   
1    A man's simple lie to secure his job escalates...   
2    A barber seeks vengeance after his home is bur...   
3    A common man's struggles against a corrupt pol...   
4    This final installment in Satyajit Ray's Apu T...   
..                                                 ...   
245  A poor but big-hearted man takes orphans into ...   
246  Set in the backdrop of Indian Emergency 1975, ...   
247  Velu, an aspiring kabaddi player, goes to Madu...   
248  Advocate Mukundan Unni, played by Vineeth Sree...   
249  A group of eight youngsters from an engineerin...   

                    Streaming platform Rating Category  \
0                              SonyLIV            Good   
1    Amazon Prime Video, YouTube, Zee5            Good   
2                              Netflix            Good   
3          Amazon Prime Video, YouTube            Good   
4          Amazon Prime Video, Hoichoi            Good   
..                                 ...    